akka-http2-support/src/main/scala/akka/http/impl/engine/http2/ProtocolSwitch.scala

```diff
@@ -135,4 +134,10 @@ private[http] object ProtocolSwitch {
         }
       }
     )
+
+  def byPreface(http1Stack: HttpServerFlow, http2Stack: HttpServerFlow): HttpServerFlow = {
+    def chooseProtocol(sessionBytes: SessionBytes): String =
+      if (sessionBytes.bytes.startsWith(Http2Protocol.ClientConnectionPreface)) "h2" else "http/1.1"
```

Comment:
@raboof Do we know that we have enough bytes here?

Reply:
Currently we don't - this PR only adds some tests (including an ignored one for that very scenario), but doesn't fix it.

Reply:
Ok!